### ISSUE TITLE
Add a "Patterns" section to the documentation

### ISSUE THREAD

**Issue by titouanmathis** [studiometa/ui#62]
The [Cloudscape design system](https://cloudscape.design/) has a ["Patterns" section](https://cloudscape.design/patterns/patterns/overview/) which contains examples on how to combine components to implement common interfaces. 

We could maybe add a similar section in our doc which would present ways of doing things with existing components.

- How to handle forms
- How to handle filters 
- How to manage loading states
- ...

Any thoughts on the concept @studiometa/core?

**Comment by perruche** [comment]
Yes this will be a great addition to have more concrete exemples.

Some more ideas:
- How to extend existing components (JS & Twig)
- How to think & compose a custom complex component
- Some more abstract component might need a use-case section to explain when/why use them

**Comment by notjb** [comment]
Really down for this, I think the most needed is a small explainer + code snippet for extending existing components  🙂

**Comment by titouanmathis** [comment]
This is addressed by [demos](https://ui.studiometa.dev/-/demos/).
